tokenizers/src/tokenizer/encoding.rs

```diff
@@ -306,78 +308,75 @@ impl Encoding {
+            type_ids: self.type_ids[start..stop].to_vec(),
+            tokens: self.tokens[start..stop].to_vec(),
+            words: self.words[start..stop].to_vec(),
+            offsets: self.offsets[start..stop].to_vec(),
+            special_tokens_mask: self.special_tokens_mask[start..stop].to_vec(),
+            attention_mask: self.attention_mask[start..stop].to_vec(),
+            overflowing: vec![],
+            sequence_ranges: HashMap::new(),
+        };
 
         loop {
-            if part_size * part_id >= o_ids.len() {
+            if i == parts_ranges.len() - 1 {
                 break;
             }
-
-            let o = Encoding {
-                ids: get_current_part(&prev_encoding.ids, &o_ids, part_size, part_id, stride),
-                type_ids: get_current_part(
-                    &prev_encoding.type_ids,
-                    &o_type_ids,
-                    part_size,
-                    part_id,
-                    stride,
-                ),
-                tokens: get_current_part(
-                    &prev_encoding.tokens,
-                    &o_tokens,
-                    part_size,
-                    part_id,
-                    stride,
-                ),
-                words: get_current_part(&prev_encoding.words, &o_words, part_size, part_id, stride),
-                offsets: get_current_part(
-                    &prev_encoding.offsets,
-                    &o_offsets,
-                    part_size,
-                    part_id,
-                    stride,
-                ),
-                special_tokens_mask: get_current_part(
-                    &prev_encoding.special_tokens_mask,
-                    &o_spe_toks,
-                    part_size,
-                    part_id,
-                    stride,
-                ),
-                attention_mask: get_current_part(
-                    &prev_encoding.attention_mask,
-                    &o_attent,
-                    part_size,
-                    part_id,
-                    stride,
-                ),
+            i += 1;
+            let (start, stop) = parts_ranges[i];
+            new_encoding.overflowing.push(Encoding {
+                ids: self.ids[start..stop].to_vec(),
+                type_ids: self.type_ids[start..stop].to_vec(),
+                tokens: self.tokens[start..stop].to_vec(),
+                words: self.words[start..stop].to_vec(),
+                offsets: self.offsets[start..stop].to_vec(),
+                special_tokens_mask: self.special_tokens_mask[start..stop].to_vec(),
+                attention_mask: self.attention_mask[start..stop].to_vec(),
                 overflowing: vec![],
                 sequence_ranges: HashMap::new(),
-            };
-
-            part_id += 1;
-            overflowing.push(o);
-            prev_encoding = overflowing.last().unwrap();
+            });
         }
-
-        self.overflowing = overflowing;
+        std::mem::swap(self, &mut new_encoding);
```

Comment:
Cannot assign to immutable value self or new_encoding's lifetime is not long enough.

I'm actually going to use replace instead of swap, I don't want to add an extra copy.

Reply:
I think it's pretty clean, what's worrying you ?

Reply:
It's definitely not bad. Just a taste for move semantics if they were available.

It seems to be working though:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=818009898768c05851b013ae7d17403d

(Again this is very personal, definitely nothing string against swap)

Reply:
Ah what an idiot, I was dereferencing the `new_encoder` and not self 😑